File: week-05/day-05/garden-again/plant.ts
'use strict'

export abstract class Plant {
  protected color: string
  protected currentWaterAm: number;
  protected absorbingCap: number;

  constructor(color: string) {
    this.color = color
    this.currentWaterAm = 0;
    this.absorbingCap = null;
  }
  abstract needWater(): void;

  public beWatered(waterAm: number): void {
    this.currentWaterAm+=waterAm*this.absorbingCap
  }

  getColor(){
    return this.color
  }

  abstract getType(): void 

}

File: datascience/week-01/garden/garden_py/plant.py
import random
import sys
import os


class Plants:

    def __init__(self, color):
        self._color = color
        self._currentWaterAm = 0
        self._absorbingCap = None

    def set_color(self, color):
        self._color = color

    def get_color(self):
        return self._color

    def needWater(self):
        pass

    def beWatered(self, waterAmount):
        self._currentWaterAm += waterAmount*self._absorbingCap

    def get_type(self):
        pass
